File: src/main/java/com/trustline/trustline/appuser/model/VerificationModel.java
package com.trustline.trustline.appuser.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.util.UUID;

@Getter
@Setter
@ToString
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "verifications")
public class VerificationModel extends AuditModel {
    @Id
    @JdbcTypeCode(SqlTypes.UUID)
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;
    private String messageId;
    private UUID userId;
    private String pin;
    @Enumerated(EnumType.STRING)
    private OtpModeEnum mode;

}

File: src/main/kotlin/trustline/appuser/model/VerificationModel.kt
package trustline.appuser.model;

import jakarta.persistence.*
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes
import trustline.appuser.dto.OtpModeEnum
import trustline.appuser.dto.VerificationType
import java.util.*


@Entity
@Table(name = "verifications")
data class VerificationModel(
    @Id
    @JdbcTypeCode(SqlTypes.UUID)
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: UUID? = null,
    val messageId: String? = null,
    val userId: UUID? = null,
    val pin: String? = null,
    @Enumerated(EnumType.STRING)
    val mode: OtpModeEnum? = null,
    @Enumerated(EnumType.STRING)
    val type: VerificationType? = null
) : AuditModel()
